In [1]:
import pandas as pd

data_ev=pd.read_json("Script_generador_tickets_sinteticos_electricos/data/tickets_ev_sinteticos.json")

In [ ]:
data_ev


idTicket idEmpresa       empresaNombre  idUsuario fechaEmision  \
0    T-54D030ADD887     EMP04  Transporte_04 S.L.  EMP04-U28   2025-04-13   
1    T-AE79ABE16820     EMP04  Transporte_04 S.L.  EMP04-U28   2025-07-18   
2    T-4CB2DB9F9164     EMP04  Transporte_04 S.L.  EMP04-U28   2025-01-12   
3    T-D9B00EC3033E     EMP04  Transporte_04 S.L.  EMP04-U28   2025-06-30   
4    T-129E5743B869     EMP04  Transporte_04 S.L.  EMP04-U28   2025-04-12   
..              ...       ...                 ...        ...          ...   
445  T-F9F262A7F356     EMP06  Transporte_06 S.L.  EMP06-U36   2025-01-15   
446  T-1B9D31F2AD64     EMP06  Transporte_06 S.L.  EMP06-U36   2025-01-22   
447  T-F182F2C3FC81     EMP06  Transporte_06 S.L.  EMP06-U36   2025-04-29   
448  T-D1CAF4D299FC     EMP06  Transporte_06 S.L.  EMP06-U36   2025-02-26   
449  T-F989356A185C     EMP06  Transporte_06 S.L.  EMP06-U36   2025-05-19   

    horaEmision          metodoPago  \
0      04:23:15  Tarjeta de empresa   
1      18:28:03  Tarjeta de empresa   
2      16:21:40  Tarjeta de empresa   
3      06:55:20  Tarjeta de empresa   
4      14:23:10  Tarjeta de empresa   
..          ...                 ...   
445    10:19:20  Tarjeta de empresa   
446    06:41:02  Tarjeta de empresa   
447    08:35:42  Tarjeta de empresa   
448    14:13:37  Tarjeta de empresa   
449    12:41:43  Tarjeta de empresa   

                                              estacion  \
0    {'id': 'ES*WEN*ESGASVALDEMORO14', 'provincia':...   
1    {'id': 'ES*IBD*E157199', 'provincia': 'TARRAGO...   
2    {'id': 'ES*EDP*E00923', 'provincia': 'Principa...   
3    {'id': 'ES*ESX*E21XP22T3KKJAK006051', 'provinc...   
4    {'id': 'ES*EVC*E14765', 'provincia': 'Navarra'...   
..                                                 ...   
445  {'id': 'ES*REP*E13476*1', 'provincia': 'Comuni...   
446  {'id': 'ES*IBD*E397991', 'provincia': 'VALENCI...   
447  {'id': 'ES*ENE*E044', 'provincia': 'Coruña, A'...   
448  {'id': 'ES*MOE*EC8A233S023CNWAUQEH', 'provinci...   
449  {'id': 'ES*ESX*E619460527800051', 'provincia':...   

                                                lineas  baseImponible   iee  \
0    [{'producto': 'Electricidad', 'kwh': 76.92, 'p...          41.08  2.10   
1    [{'producto': 'Electricidad', 'kwh': 44.94, 'p...          24.99  1.28   
2    [{'producto': 'Electricidad', 'kwh': 45.55, 'p...          24.41  1.25   
3    [{'producto': 'Electricidad', 'kwh': 63.51, 'p...          35.18  1.80   
4    [{'producto': 'Electricidad', 'kwh': 46.32, 'p...          25.43  1.30   
..                                                 ...            ...   ...   
445  [{'producto': 'Electricidad', 'kwh': 41.79, 'p...          22.61  1.16   
446  [{'producto': 'Electricidad', 'kwh': 32.3, 'pr...          17.64  0.90   
447  [{'producto': 'Electricidad', 'kwh': 33.68, 'p...          18.29  0.93   
448  [{'producto': 'Electricidad', 'kwh': 43.81, 'p...          23.66  1.21   
449  [{'producto': 'Electricidad', 'kwh': 41.24, 'p...          21.94  1.12   

      iva  total moneda         tipoDocumento  
0    9.06  52.24    EUR  Factura simplificada  
1    5.51  31.78    EUR  Factura simplificada  
2    5.39  31.05    EUR  Factura simplificada  
3    7.77  44.75    EUR  Factura simplificada  
4    5.61  32.34    EUR  Factura simplificada  
..    ...    ...    ...                   ...  
445  4.98  28.75    EUR  Factura simplificada  
446  3.89  22.43    EUR  Factura simplificada  
447  4.04  23.26    EUR  Factura simplificada  
448  5.22  30.09    EUR  Factura simplificada  
449  4.84  27.90    EUR  Factura simplificada  

[450 rows x 15 columns]

In [3]:
data_ev["lineas"]

0      [{'producto': 'Electricidad', 'kwh': 76.92, 'p...
1      [{'producto': 'Electricidad', 'kwh': 44.94, 'p...
2      [{'producto': 'Electricidad', 'kwh': 45.55, 'p...
3      [{'producto': 'Electricidad', 'kwh': 63.51, 'p...
4      [{'producto': 'Electricidad', 'kwh': 46.32, 'p...
                             ...                        
445    [{'producto': 'Electricidad', 'kwh': 41.79, 'p...
446    [{'producto': 'Electricidad', 'kwh': 32.3, 'pr...
447    [{'producto': 'Electricidad', 'kwh': 33.68, 'p...
448    [{'producto': 'Electricidad', 'kwh': 43.81, 'p...
449    [{'producto': 'Electricidad', 'kwh': 41.24, 'p...
Name: lineas, Length: 450, dtype: object

In [4]:
estacion_df = pd.json_normalize(data_ev["estacion"])
data_ev = data_ev.drop(columns=["estacion"]).join(estacion_df.add_prefix("estacion."))


In [5]:
df_exploded = data_ev.explode("lineas").reset_index(drop=True)

# Normaliza la columna "lineas" (dict → columnas)
lineas_normalizadas = pd.json_normalize(df_exploded["lineas"])

# Une las nuevas columnas con el dataset original
data_ev = df_exploded.drop(columns=["lineas"]).join(lineas_normalizadas.add_prefix("lineas."))

In [6]:
import datetime
def hora(x):
    datetime.strptime()

In [7]:
data_ev["fechaEmision"]=pd.to_datetime(data_ev["fechaEmision"])

data_ev["horaEmision"]=pd.to_datetime(data_ev["horaEmision"], format="%H:%M:%S")

In [8]:
data_ev["fechaEmision"]

0     2025-04-13
1     2025-07-18
2     2025-01-12
3     2025-06-30
4     2025-04-12
         ...    
445   2025-01-15
446   2025-01-22
447   2025-04-29
448   2025-02-26
449   2025-05-19
Name: fechaEmision, Length: 450, dtype: datetime64[ns]

In [9]:
data_ev["horaEmision"]

0     1900-01-01 04:23:15
1     1900-01-01 18:28:03
2     1900-01-01 16:21:40
3     1900-01-01 06:55:20
4     1900-01-01 14:23:10
              ...        
445   1900-01-01 10:19:20
446   1900-01-01 06:41:02
447   1900-01-01 08:35:42
448   1900-01-01 14:13:37
449   1900-01-01 12:41:43
Name: horaEmision, Length: 450, dtype: datetime64[ns]

In [10]:
data_ev.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 450 entries, 0 to 449
Data columns (total 28 columns):
 #   Column                             Non-Null Count  Dtype         
---  ------                             --------------  -----         
 0   idTicket                           450 non-null    object        
 1   idEmpresa                          450 non-null    object        
 2   empresaNombre                      450 non-null    object        
 3   idUsuario                          450 non-null    object        
 4   fechaEmision                       450 non-null    datetime64[ns]
 5   horaEmision                        450 non-null    datetime64[ns]
 6   metodoPago                         450 non-null    object        
 7   baseImponible                      450 non-null    float64       
 8   iee                                450 non-null    float64       
 9   iva                                450 non-null    float64       
 10  total                              450

In [11]:
data_ev['empresaNombre']


0      Transporte_04 S.L.
1      Transporte_04 S.L.
2      Transporte_04 S.L.
3      Transporte_04 S.L.
4      Transporte_04 S.L.
              ...        
445    Transporte_06 S.L.
446    Transporte_06 S.L.
447    Transporte_06 S.L.
448    Transporte_06 S.L.
449    Transporte_06 S.L.
Name: empresaNombre, Length: 450, dtype: object

In [12]:
[['empresaNombre', 'idUsuario', 'fechaEmision', 'horaEmision', 'metodoPago', 'baseImponible', 'total', 'estacion.id', 'estacion.provincia']]

[['empresaNombre',
  'idUsuario',
  'fechaEmision',
  'horaEmision',
  'metodoPago',
  'baseImponible',
  'total',
  'estacion.id',
  'estacion.provincia']]

In [13]:
from skrub import TableReport

TableReport(data_ev) 

Processing column  28 / 28


<TableReport: use .open() to display>

* No parceqeue haya diferencias significativas entre la hora y la 
* Hay tres empresas y tres emplados por empresa
* El metodo de pago mas comuns es la tarjete de empresa
* Las fechas van desde enero a 30 julio
* Lo normal es gastar 28 euros en electriduad (unos 41.8 kwh)
* Barcelnoa Madris son los que tienen mas gpuesto
* Por algun mtotivo esta estacion 'Parking Publico Ciudad Imagen - C. Luis' ha vendio casi el doble que el resto
* el IVA varia bastante mas de lo esperado

In [14]:
data_ev["idEmpresa"].value_counts()

idEmpresa
EMP04    150
EMP05    150
EMP06    150
Name: count, dtype: int64

In [15]:
data_ev.fechaEmision.max()

Timestamp('2025-07-30 00:00:00')

In [16]:
data_ev['estacion.empresa'].value_counts()

estacion.empresa
ENDESA X WAY S.L.                                90
IBERDROLA CLIENTES S.A.U                         85
REPSOL SOLUCIONES ENERGÉTICAS SA                 45
Etecnic Movilidad Electrica SRL                  40
REPSOL COMERCIAL DE PRODUCTOS PETROLIFEROS SA    34
WENEA EMOBILITY SERVICES S.L.                    23
EDP CLIENTES SAU                                 16
CHARGING TOGETHER SL                             13
ERANOVUM E-MOBILITY S.L.                         11
PLENOIL S.L.                                     10
EXANERGÍA S.L.                                    8
ACCIONA Recarga                                   5
FENIE ENERGIA SA                                  4
Prio Recharge Sl.                                 4
BARCELONA DE SERVEIS MUNICIPALS SA                4
(ZUNDER) Grupo Easychargar SA                     4
IONITY GmbH Sucursal en España                    4
ION SMART ENERGY 2000 SL                          3
Tesla Spain SLU                                

In [17]:
data_ev['lineas.producto'].value_counts()

lineas.producto
Electricidad    450
Name: count, dtype: int64

In [18]:
import plotly.graph_objs as go
from plotly.offline import init_notebook_mode, iplot
data_ev.groupby('estacion.empresa')["lineas.precioUnitario"].median()



# Creating trace1
trace1 = go.Bar(
                    x = data_ev['estacion.empresa'],
                    y = data_ev['lineas.precioUnitario'],
                    name = 'precio unitacio',
                    
                    
                    )

fig = go.Figure(data = trace1)

iplot(fig)

Iberdrola y endesa son los principales vendedores (aunque parce que varioan mucho los precion seguramente porque son los que mas venden)
Repsol tampbien vende mucho aunque este divido en dos subcompañias (una dirigiaa gasolina y otra a energia)
Wenea EDP Etecnic y Charging togetehe 

In [19]:
trace1

Bar({
    'name': 'precio unitacio',
    'x': array(['WENEA EMOBILITY SERVICES S.L.', 'IBERDROLA CLIENTES S.A.U',
                'EDP CLIENTES SAU', ..., 'ENERGEA CONSULTING Y ASESORES SL',
                'CEPSA COMERCIAL PETRÓLEOS.A.U.', 'ENDESA X WAY S.L.'], dtype=object),
    'y': array([0.679, 0.707, 0.682, ..., 0.691, 0.687, 0.677])
})

In [20]:
data_ev.groupby('lineas.kwh')["lineas.precioUnitario"].median()


trace1 = go.Scatter(
                    x = data_ev['lineas.kwh'],
                    y = data_ev['lineas.precioUnitario'],
                    name = 'precio unitacio',
                    mode = 'markers',


                    
                    
                    )
layout=go.Layout(
            xaxis= dict(title= 'kwh',ticklen= 5,zeroline= False),
              yaxis= dict(title= 'precioUnitario',ticklen= 5,zeroline= False)
)
fig = go.Figure(data = trace1)
iplot(fig)

Esto es interesante: hay dos dos tipos de precios pero no es por kwh

In [21]:
data_ev.groupby('lineas.kwh')["lineas.precioUnitario"].median()


trace1 = go.Scatter(
                    x = data_ev['estacion.tarifa'],
                    y = data_ev['lineas.precioUnitario'],
                    name = 'precio unitacio',
                    mode = 'markers',


                    
                    
                    )
layout=go.Layout(
            xaxis= dict(title= 'tarifa',ticklen= 5,zeroline= False),
              yaxis= dict(title= 'precioUnitario',ticklen= 5,zeroline= False)
)
fig = go.Figure(data = trace1)
iplot(fig)

La diferencia de precios es si es de corriente directa o alterna (se supono que DC es mas rapido que AC po)

In [22]:
data_ev.groupby('estacion.empresa')["lineas.precioUnitario"].median()

estacion.empresa
(ZUNDER) Grupo Easychargar SA                    0.7005
ACCIONA Recarga                                  0.6970
AJUNTAMENT DE CALVIÀ                             0.7000
AJUNTAMENT DES CASTELL                           0.6880
AYUNTAMIENTO DE FELANITX                         0.5320
Alcudiamar S.L.                                  0.6870
Allego Spain S.L.U                               0.7060
Atlante Iberia SL                                0.5190
BARCELONA DE SERVEIS MUNICIPALS SA               0.5510
BP Energía España SAU                            0.6920
CENTROS COMERCIALES CARREFOUR SA                 0.6820
CEPSA COMERCIAL PETRÓLEOS.A.U.                   0.7060
CHARGING TOGETHER SL                             0.7050
Consejo Insular de la Energía de Gran Canaria    0.7010
EDP CLIENTES SAU                                 0.6735
EFICIENCIA Y SERVICIOS RENOVABLES SL             0.5270
ENDESA X WAY S.L.                                0.6890
ENERGEA CONSULTING Y ASESORES S

Lo mas barato es el gasoleo A

In [23]:
import folium
from folium import plugins
mapa = folium.Map(location=[40, -5], zoom_start=6, )
for x in range(len(data_ev)):
         
    
    folium.Marker(location=data_ev.iloc[x][["estacion.lat","estacion.lon"]], popup=data_ev.iloc[x]["lineas.precioUnitario"]
              
              ).add_to(mapa)

mapa


c:\Users\unaif\anaconda3\envs\venv\Lib\site-packages\folium\utilities.py:95: FutureWarning:

Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`

c:\Users\unaif\anaconda3\envs\venv\Lib\site-packages\folium\utilities.py:95: FutureWarning:

Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`

c:\Users\unaif\anaconda3\envs\venv\Lib\site-packages\folium\utilities.py:95: FutureWarning:

Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`

c:\Users\unaif\anaconda3\envs\venv\Lib\site-packages\folium\utiliti

In [24]:
['idEmpresa', 'idUsuario']

['idEmpresa', 'idUsuario']

In [25]:
data_ev.groupby('idEmpresa')['idUsuario'].value_counts()

idEmpresa  idUsuario
EMP04      EMP04-U28    50
           EMP04-U29    50
           EMP04-U30    50
EMP05      EMP05-U31    50
           EMP05-U32    50
           EMP05-U33    50
EMP06      EMP06-U34    50
           EMP06-U35    50
           EMP06-U36    50
Name: count, dtype: int64

Todos los usuarisos compranrpn el mismp numero de veces

In [26]:
data_ev.groupby('idEmpresa')[['lineas.importe']].mean()

lineas.importe
idEmpresa                
EMP04           27.865533
EMP05           28.163000
EMP06           27.810667

In [27]:
data_ev.groupby('idEmpresa')['estacion.tarifa'].value_counts()

idEmpresa  estacion.tarifa
EMP04      DC                 124
           AC                  26
EMP05      DC                 124
           AC                  26
EMP06      DC                 119
           AC                  31
Name: count, dtype: int64

In [28]:
data_ev.columns

Index(['idTicket', 'idEmpresa', 'empresaNombre', 'idUsuario', 'fechaEmision',
       'horaEmision', 'metodoPago', 'baseImponible', 'iee', 'iva', 'total',
       'moneda', 'tipoDocumento', 'estacion.id', 'estacion.provincia',
       'estacion.municipio', 'estacion.direccion', 'estacion.lat',
       'estacion.lon', 'estacion.empresa', 'estacion.nifEmpresa',
       'estacion.potenciaMaxKW', 'estacion.tarifa', 'lineas.producto',
       'lineas.kwh', 'lineas.precioUnitarioSinImpuestos',
       'lineas.precioUnitario', 'lineas.importe'],
      dtype='object')

Todas las empresas pagan parecidas y tienen poblaciones parecidas en todos 

In [29]:
display(data_ev.groupby('idEmpresa')['fechaEmision'].value_counts())

idEmpresa  fechaEmision
EMP04      2025-01-02      3
           2025-02-04      3
           2025-02-16      3
           2025-02-24      3
           2025-02-27      3
                          ..
EMP06      2025-07-19      1
           2025-07-20      1
           2025-07-21      1
           2025-07-24      1
           2025-07-30      1
Name: count, Length: 320, dtype: int64

In [30]:

from folium.plugins import HeatMap
df_map = data_ev.groupby(['estacion.lat', 'estacion.lon'])['lineas.kwh'].sum().reset_index()

# Crear mapa centrado (ejemplo: centro de España)
m = folium.Map(location=[40.0, -3.7], zoom_start=6, tiles="CartoDB positron")

# Preparar datos para el HeatMap (lat, lon, peso)
heat_data = [[row['estacion.lat'], row['estacion.lon'], row['lineas.kwh']] for index, row in df_map.iterrows()]

# Añadir capa HeatMap
HeatMap(heat_data, radius=15, blur=10, max_zoom=1).add_to(m)



# Mostrar en Jupyter
print(m.to_json())

{"name": "Map", "id": "6ba113058453b5bb74c3f804ec2af3f3", "children": {"cartodbpositron": {"name": "TileLayer", "id": "b8c763fd67122700f8167d006eae07b4", "children": {}}, "heat_map_c50e9e915d77ae36e066e02a397a4e2e": {"name": "HeatMap", "id": "c50e9e915d77ae36e066e02a397a4e2e", "children": {}}}}


In [31]:
import json
m = folium.Map(location=[40.0, -3.7], zoom_start=6, tiles="CartoDB positron")

# Preparar datos para el HeatMap (lat, lon, peso)
heat_data = [[row['estacion.lat'], row['estacion.lon'], row['lineas.kwh']] for index, row in df_map.iterrows()]

# Añadir capa HeatMap
HeatMap(heat_data, radius=15, blur=10, max_zoom=1).add_to(m)

# --- Guardar los datos en JSON ---
"""with open("heatmap_data.json", "w", encoding="utf-8") as f:
    json.dump(heat_data, f, ensure_ascii=False, indent=2)"""
print(heat_data)

[[np.float64(27.7679069), np.float64(-15.6011738), np.float64(22.5)], [np.float64(27.795289), np.float64(-15.704981), np.float64(43.19)], [np.float64(27.8191842), np.float64(-15.763147), np.float64(50.83)], [np.float64(28.009833), np.float64(-15.390083), np.float64(43.81)], [np.float64(28.0529798), np.float64(-16.7152284), np.float64(32.5)], [np.float64(28.0962706), np.float64(-15.415256), np.float64(57.1)], [np.float64(28.1143065), np.float64(-15.4487773), np.float64(23.59)], [np.float64(28.115359), np.float64(-15.420044), np.float64(37.11)], [np.float64(28.128655), np.float64(-15.450266), np.float64(32.21)], [np.float64(28.178401), np.float64(-17.265379), np.float64(30.05)], [np.float64(28.415429), np.float64(-16.542907), np.float64(51.25)], [np.float64(28.4591687), np.float64(-16.256747), np.float64(41.57)], [np.float64(28.496343), np.float64(-13.863596), np.float64(25.79)], [np.float64(28.496423), np.float64(-13.863676), np.float64(55.14)], [np.float64(28.6108129), np.float64(-13.9

In [32]:
print(HeatMap(heat_data, radius=15, blur=10, max_zoom=1).to_json())

{"name": "HeatMap", "id": "2cf9536599661f70da4c7343f797eaa1", "children": {}}


In [141]:

from folium.plugins import HeatMap
df_map = data_ev.groupby(['estacion.lat', 'estacion.lon'])['lineas.importe'].sum().reset_index()

# Crear mapa centrado (ejemplo: centro de España)
m = folium.Map(location=[40.0, -3.7], zoom_start=6, tiles="CartoDB positron")

# Preparar datos para el HeatMap (lat, lon, peso)
heat_data = [[row['estacion.lat'], row['estacion.lon'], row['lineas.importe']] for index, row in df_map.iterrows()]

# Añadir capa HeatMap
HeatMap(heat_data, radius=15, blur=10, max_zoom=1).add_to(m)

# Mostrar en Jupyter
m

No parece que haya diferencias entre importe y el precio total.
Tambien parece que Depende mas del numero de estacioes que del o gasto

In [147]:
df_map = data_ev[data_ev['estacion.tarifa']=="AC"].groupby(['estacion.lat', 'estacion.lon'])['estacion.tarifa'].count().reset_index()

# Crear mapa centrado (ejemplo: centro de España)
m = folium.Map(location=[40.0, -3.7], zoom_start=6, tiles="CartoDB positron")

# Preparar datos para el HeatMap (lat, lon, peso)
heat_data = [[row['estacion.lat'], row['estacion.lon'], row['estacion.tarifa']] for index, row in df_map.iterrows()]

# Añadir capa HeatMap
HeatMap(heat_data, radius=15, blur=10, max_zoom=1).add_to(m)

# Mostrar en Jupyter
m

LA localizacion es algo distinta de DC y AC, aunque seguramente es porque hay menos tickect de AC

In [151]:
data_ev.groupby('lineas.kwh')["lineas.precioUnitario"].median()


trace1 = go.Scatter(
                    x = data_ev[data_ev['estacion.tarifa']=="AC"]['fechaEmision'],
                    y = data_ev[data_ev['estacion.tarifa']=="AC"]['lineas.precioUnitario'],
                    name = 'AC',
                    mode = 'markers',)


trace2 = go.Scatter(
                    x = data_ev[data_ev['estacion.tarifa']=="DC"]['fechaEmision'],
                    y = data_ev[data_ev['estacion.tarifa']=="DC"]['lineas.precioUnitario'],
                    name = 'DC',
                    mode = 'markers',)

layout=go.Layout(
            xaxis= dict(title= 'fecha',ticklen= 5,zeroline= False),
              yaxis= dict(title= 'precioUnitario',ticklen= 5,zeroline= False)
)
fig = go.Figure(data = [trace1, trace2])
iplot(fig)

No hay no hay estacionalidad en el tiempo

No hay diferecencia entre empresa

In [164]:
trace1 = go.Scatter(
                    x = data_ev[data_ev['idUsuario']=="EMP04-U28"]['fechaEmision'],
                    y = data_ev[data_ev['idUsuario']=="EMP04-U28"]['lineas.precioUnitario'],
                    name = 'EMP04-U28',
                    mode = 'markers',)

trace2 = go.Scatter(
                    x = data_ev[data_ev['idUsuario']=="EMP04-U29"]['fechaEmision'],
                    y = data_ev[data_ev['idUsuario']=="EMP04-U29"]['lineas.precioUnitario'],
                    name = 'EMP04-U29',
                    mode = 'markers',)
trace3 = go.Scatter(
                    x = data_ev[data_ev['idUsuario']=="EMP04-U30"]['fechaEmision'],
                    y = data_ev[data_ev['idUsuario']=="EMP04-U30"]['lineas.precioUnitario'],
                    name = 'EMP04-U30',
                    mode = 'markers',)

trace4 = go.Scatter(
                    x = data_ev[data_ev['idUsuario']=="EMP05-U31"]['fechaEmision'],
                    y = data_ev[data_ev['idUsuario']=="EMP05-U31"]['lineas.precioUnitario'],
                    name = 'EMP05-U31',
                    mode = 'markers',)

trace5 = go.Scatter(
                    x = data_ev[data_ev['idUsuario']=="EMP05-U32"]['fechaEmision'],
                    y = data_ev[data_ev['idUsuario']=="EMP05-U32"]['lineas.precioUnitario'],
                    name = 'EMP05-U32',
                    mode = 'markers',)

trace6 = go.Scatter(
                    x = data_ev[data_ev['idUsuario']=="EMP05-U33"]['fechaEmision'],
                    y = data_ev[data_ev['idUsuario']=="EMP05-U33"]['lineas.precioUnitario'],
                    name = 'EMP05-U33',
                    mode = 'markers',)

trace7 = go.Scatter(
                    x = data_ev[data_ev['idUsuario']=="EMP06-U34"]['fechaEmision'],
                    y = data_ev[data_ev['idUsuario']=="EMP06-U34"]['lineas.precioUnitario'],
                    name = 'EMP06-U34',
                    mode = 'markers',)

trace8 = go.Scatter(
                    x = data_ev[data_ev['idUsuario']=="EMP06-U35"]['fechaEmision'],
                    y = data_ev[data_ev['idUsuario']=="EMP06-U35"]['lineas.precioUnitario'],
                    name = 'EMP06-U35',
                    mode = 'markers',)
trace9 = go.Scatter(
                    x = data_ev[data_ev['idUsuario']=="EMP06-U36"]['fechaEmision'],
                    y = data_ev[data_ev['idUsuario']=="EMP06-U36"]['lineas.precioUnitario'],
                    name = 'EMP06-U36',
                    mode = 'markers',)

layout=go.Layout(
            xaxis= dict(title= 'fecha',ticklen= 5,zeroline= False),
              yaxis= dict(title= 'precioUnitario',ticklen= 5,zeroline= False)
)
fig = go.Figure(data = [trace1, trace2, trace3,trace4, trace5, trace6, trace7, trace8, trace9])
iplot(fig)

No hay deferencia significativa entre usuarios ni empresa